src/core/Akka/Actor/Stash/Internal/AbstractStash.cs

```diff
@@ -61,12 +63,13 @@ public void Stash()
         {
             var currMsg = _actorCell.CurrentMessage;
             var sender = _actorCell.Sender;
-            if(_theStash.Count > 0)
+
+            if (_actorCell.CurrentEnvelopeId == _currentEnvelopeId)
```

Comment:
Yeah, if you only stash in in every 4.3 billion messages... Let it crash and document the fact that it WILL crash under that scenario with a comment in the source.